Add `instanceof.unsafe` lint option (#1209)
eisop/checker-framework: docs/manual/warnings.tex

\item
  \code{cast:unsafe} (default: on) warn about unsafe casts that are not
  checked at run time, as in \code{((@NonNull String) myref)}.  Such casts

>   are generally not necessary because of type refinement

Author: `Such casts are generally not necessary` does not make sense to me because a unsafe cast should be `unsafe` instead of `not necessary`.

>   Such casts are unsafe, and they can break the property of the type system.

Reviewer: I guess the original point was that manually adding downcasts is not necessary as flow-sensitive type refinement automatically refines the type if it is safe to do so.
The "break the property" sounds a bit odd to me. Can you try to reformulate the sentence.

\item

>   \code{cast:redundant} (default: off) warn about redundant

Author: `cast:redundant` is not enabled by default based on the original code.
```
checker.getLintOption("cast:redundant", false)
```

\item
  \code{cast} Enable or disable all cast-related warnings.

\item

>   \code{instanceof:unsafe} (default: on) warn about unsafe instanceof pattern
>   that are not checked at run time, as in \code{o instanceof @Untainted String}.
>   Such instanceof patterns are unsafe, and they can break the property of the type system.

Reviewer: ```suggestion
  \code{instanceof:unsafe} (default: on) warn about unsafe instanceof tests
  that are not checked at run time, as in \code{o instanceof @Untainted String}.
  Such instanceof tests are unsafe as the type qualifiers are ignored at run time.
```

\item
  \code{cast} Enable or disable all cast-related warnings.

\item

>   \code{instanceof} Enable or disable instanceof-related warnings.

Reviewer: Add text for `instanceof:unsafe` above.
